### PR TITLE
vimPlugins.nvim-treesitter: follow up to #198606

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

- fix formatting of the generated file
- update nvim-treesitter grammars
- create aliases for `nvim-treesitter.builtGrammars` to overwrite tree-sitter grammars in `nvim-treesitter.withPlugins`
- add myself as a maintainer

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by figsoda** [comment]
there is a little too much `lib.*`s now, im wondering if i should just add an `inherit (lib)`

**Comment by SuperSandro2000** [comment]
> there is a little too much `lib.*`s now, im wondering if i should just add an `inherit (lib)`

I don't mind but if you want to change it I'd rather add a local ``with lib`` than maintaining a list of inherits.

File: pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix [hunk 1]
```diff
@@ -3,11 +3,26 @@
...
+      (k: v:
+        let
+          replaced = lib.replaceStrings [ "_" ] [ "-" ] k;
+        in
+        map (lib.flip lib.nameValuePair v)
+          ([ ("tree-sitter-${k}") ] ++ lib.optionals (k != replaced) [
+            replaced
+            "tree-sitter-${replaced}"
+          ]))
+      generatedDerivations));
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix:23]: This is generating aliases with - right? Maybe add a really short comment.

**Review by SuperSandro2000 — APPROVED** [review]

File: pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix (hunk 1 above)
Comment (figsoda) [reply pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix:23]: yes, for example, ocaml_interface will ads ocaml-interface, tree-sitter-ocaml-interface and tree-sitter-ocaml_interface

**Pull request merged**